solr/core/src/java/org/apache/solr/handler/admin/HealthCheckHandler.java

```diff
@@ -135,6 +153,80 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
     rsp.add(STATUS, OK);
   }
 
+  private void healthCheckLegacyMode(SolrQueryRequest req, SolrQueryResponse rsp) {
+    Integer maxGenerationLag = req.getParams().getInt(HealthCheckRequest.PARAM_MAX_GENERATION_LAG);
+    List<String> laggingCoresInfo = new ArrayList<>();
+    boolean allCoresAreInSync = true;
+
+    // check only if max generation lag is specified
+    if(maxGenerationLag != null) {
```

Comment:
Do we need to check for `> 0`?

Reply:
good to add the check because configurations don't come with guarantees :)